File: Api/Timesheet/Entities/Reports/EmployeeAttendanceReport.cs
using Microsoft.EntityFrameworkCore;

namespace Timesheet.Entities.Reports
{
    [Keyless]
    public class SquadAttendanceReport
    {
        public string SquadName { get; set; } = string.Empty;
        public int EmployeeId { get; set; }
        public string Name { get; set; } = string.Empty;
        public int WeekNumber { get; set; }
        public string LiteralDay { get; set; } = string.Empty;
        public DateTime? CheckInTime { get; set; }
        public DateTime? CheckInRecord { get; set; }
        public int? CheckInDifferenceInMinutes { get; set; }
        public string CheckInMessage { get; set; } = string.Empty;
        public DateTime? DepartureTime { get; set; }
        public DateTime? CheckOutRecord { get; set; }
        public int? CheckOutDifferenceInMinutes { get; set; }
        public string CheckOutMessage { get; set; } = string.Empty;
    }
}

File: Web/src/app/models/records/Reports/employee-attendance-report.ts
export class EmployeeAttendanceReport {
    EmployeeId: number;
    Email: string;
    CheckInIime: Date;
    CheckInRecord: Date;
    CheckInDifferenceInMinutes: number;
    CheckInMessage: string;
    DepartureTime: Date;
    CheckOutRecord: Date;
    CheckOutDifferenceInMinutes: number;
    CheckOutMessage: string;
    
    
  
    constructor(params?: Partial<EmployeeAttendanceReport>) {
        this.EmployeeId = params?.EmployeeId ?? 0;
        this.Email = params?.Email ?? 'no email';
        this.CheckInIime = params?.CheckInIime ?? new Date();
        this.CheckInRecord = params?.CheckInRecord  ?? new Date();
        this.CheckInDifferenceInMinutes = params?.CheckInDifferenceInMinutes ?? 1;
        this.CheckInMessage = params?.CheckInMessage ?? "no message";
        this.DepartureTime= params?.DepartureTime ?? new Date();
        this.CheckOutRecord= params?.CheckOutRecord ?? new Date();
        this.CheckOutDifferenceInMinutes = params?.CheckOutDifferenceInMinutes ?? 1;
        this.CheckOutMessage = params?.CheckOutMessage ?? "no message";
      }
  }
  